File: mirascope/experimental/v2/llm/responses/finish_reason.py
"""The reason the LLM finished generating a response."""

from enum import Enum


class FinishReason(str, Enum):
    """The reason why the LLM finished generating a response.

    TODO: add all of the finish reasons.
    """

    STOP = "stop"
    MAX_TOKENS = "max_tokens"

File: typescript/src/llm/responses/finish-reason.ts
/**
 * Finish reason indicating why the model stopped generating.
 */

/**
 * Possible reasons why generation stopped.
 */
export const FinishReason = {
  /** Model ran out of tokens. */
  MAX_TOKENS: 'max_tokens',
  /** Model refused to generate. */
  REFUSAL: 'refusal',
  /** Context length exceeded. */
  CONTEXT_LENGTH_EXCEEDED: 'context_length_exceeded',
} as const;

/**
 * Type representing a finish reason value.
 */
export type FinishReason = (typeof FinishReason)[keyof typeof FinishReason];
